test/topology_custom/test_table_desc_read_barrier.py

```diff
@@ -60,9 +61,17 @@ async def test_table_desc_read_barrier(manager: ManagerClient) -> None:
         desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
         assert desc_schema[0] != desc_schema[1]
 
-        await handler.message()
-
-        # TODO: trigger the read barrier once available
+        async def resume():
+            # extra sleep before resuming the apply
+            # (keep paused for some more time while read barrier is triggered)
+            await asyncio.sleep(timedelta(milliseconds=100).total_seconds())
+            await handler.message()
+
+        # wait for the first node to see the latest state (after the delay ends)
+        await asyncio.gather(
+            read_barrier(manager.api, servers[0].ip_addr),
+            resume()
+        )
```

Comment:
I don't see the necessity for using the handler version of `inject` at all.

Just do a sleep injection. In the test call read_barrier; it will wait for sleep to finish because it will wait for `apply` to finish.

Reply:
> I think this test is fine and we don't need a perfect deterministic solution. However, we have ways to achieve it.

In this case I wouldn't waste time trying to come up with a fully deterministic solution

I would remove the assertion
```
         assert desc_schema[0] != desc_schema[1]
```
since this is not what we want to test at the end. We want to test that the schema is synced after the read barrier, we don't want to test that the schema is not synced before it.

And then a simple sleep will work without making the test flaky. It might make it less accurate -- in theory the change may propagate naturally and test would pass without the barrier, but in most cases the barrier is still required, and that is enough to test that the barrier works.

Reply:
reverted to the previous version to just use the simple pause (increase the timeout to 1s)